File: deaf_grandma.py
def deaf_grandma():
    number_of_goodbyes = 0

    while number_of_goodbyes <= 1:
        response = input("Say something to Grandma: ")

        if response == "GOODBYE!":
            if number_of_goodbyes < 1:
                number_of_goodbyes += 1
                print("LEAVING SO SOON?")
            else:
                print("LATER SKATER!")
                return
        elif response == "":
            print("WHAT?!")
        elif response.islower():
            print("SPEAK UP, KID!")
        else:
            print("NO, NOT SINCE 1946!")



deaf_grandma()

File: deaf_grandma.js
let numberOfGoodbyes = 0;

function deafGrandma() {
  var prompt = require("prompt");
  //
  // Start the prompt
  //
  prompt.start();

  prompt.get(["userInput"], function (err, result) {
    if (result.userInput === "") {
      console.log("WHAT?!");
      deafGrandma();
    } else if (result.userInput === "GOODBYE!" && numberOfGoodbyes === 1) {
      console.log("LATER, SKATER!");
      return;
    } else if (result.userInput === "GOODBYE!") {
      console.log("LEAVING SO SOON?");
      deafGrandma();
      numberOfGoodbyes++;
    } else if (result.userInput === result.userInput.toUpperCase()) {
      console.log("NO, NOT SINCE 1946!");
      deafGrandma();
    } else {
      console.log("SPEAK UP, KID!");
      deafGrandma();
    }
  });
}

deafGrandma();
